sql/core/src/test/scala/org/apache/spark/sql/execution/adaptive/AdaptiveQueryExecSuite.scala

```diff
@@ -39,6 +40,20 @@ class AdaptiveQueryExecSuite
   setupTestData()
 
   private def runAdaptiveAndVerifyResult(query: String): (SparkPlan, SparkPlan) = {
+    var finalPlanExists = false
+    val listener = new SparkListener {
+      override def onOtherEvent(event: SparkListenerEvent): Unit = {
+        event match {
+          case SparkListenerSQLAdaptiveExecutionUpdate(_, _, sparkPlanInfo) =>
+            if (sparkPlanInfo.simpleString.startsWith("AdaptiveSparkPlan(isFinalPlan=true)")) {
```

Comment:
can we make it more type safe? e.g. check if it's a `AdaptiveSparkPlanExec`, and check the `isFinalPlan` flag.

Reply:
`SparkPlanInfo` has lost the type info. Any other way to check ?